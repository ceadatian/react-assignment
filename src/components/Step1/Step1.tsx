/**
 * requirement:
 * a dropdown menu for selecting the meal category 
 * a number input field for the number of people.
 * 
 */

import React, { useState } from 'react';
import { Form, Select, InputNumber, Button } from 'antd';
import './Step1.css'

interface Props {
  onSubmit: (mealCategory: string, numberOfPeople: number) => void;
}

const Step1: React.FC<Props> = ({ onSubmit }) => {
  const [mealCategory, setMealCategory] = useState<string>('breakfast');
  const [numberOfPeople, setNumberOfPeople] = useState<number>(0);

  // const handleMealCategoryChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
  //   setMealCategory(event.target.value);
  // };

  // const handleNumberOfPeopleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
  //   setNumberOfPeople(Number(event.target.value));
  // };

  const handleNumberOfPeopleChange = (value: number | null) => {
    if (value !== null) {
      setNumberOfPeople(value);
    }
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    // event.preventDefault();
    onSubmit(mealCategory, numberOfPeople);
  };

  return (
    <Form className='global-step' onFinish={handleSubmit}>
      <Form.Item
        rules={[{ required: true, message: 'Please Select a meal' }]}
        label="Please Select a meal:"
        style={{ width: '300px'}}
      >
        <Select value={mealCategory} onChange={setMealCategory}>
          <Select.Option value="breakfast">Breakfast</Select.Option>
          <Select.Option value="lunch">Lunch</Select.Option>
          <Select.Option value="dinner">Dinner</Select.Option>
        </Select>
      </Form.Item>
      <Form.Item
        rules={[{ required: true}]}
        label="Please Enter Number of People:"
        style={{ width: '600px'}}
      >
        <InputNumber
          min={1}
          max={10}
          value={numberOfPeople}
          onChange={handleNumberOfPeopleChange}
          required
        />
      </Form.Item>
      <Form.Item>
        <Button type="primary" htmlType="submit" disabled={!mealCategory || !numberOfPeople}>Next</Button>
      </Form.Item>
    </Form>
  );
};

export default Step1;
