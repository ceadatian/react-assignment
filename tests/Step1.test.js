import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Step1 from "../src/components/Step1/Step1";

configure({ adapter: new Adapter() });
describe('Step1 component', () => {
  let wrapper;
  const mockOnSubmit = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Step1 onSubmit={mockOnSubmit} />);
  });

  it('renders without crashing', () => {
    expect(wrapper).toBeTruthy();
  });

  it('renders a Select component', () => {
    expect(wrapper.find('#ant-select')).toHaveLength(1);
  });

  it('renders a InputNumber component', () => {
    expect(wrapper.find('InputNumber')).toHaveLength(1);
  });

  it('renders a Button component', () => {
    expect(wrapper.find('Button')).toHaveLength(1);
  });

  it('calls the onSubmit function when the form is submitted', () => {
    wrapper.find('#ant-form1').simulate('finish');
    expect(mockOnSubmit).toHaveBeenCalled();
  });

  it('calls the onSubmit function with the correct arguments', () => {
    wrapper.find('#ant-select').simulate('change', 'lunch');
    wrapper.find('InputNumber').simulate('change', 5);
    wrapper.find('#ant-form1').simulate('finish');
    expect(mockOnSubmit).toHaveBeenCalledWith('lunch', 5);
  });

  it('disables the button when the form is not filled out', () => {
    expect(wrapper.find('Button').prop('disabled')).toBeTruthy();
  });

  it('enables the button when the form is filled out', () => {
    wrapper.find('#ant-select').simulate('change', { target: { value: 'lunch' } });
    wrapper.find('InputNumber').simulate('change', { target: { value: 5 } });
    expect(wrapper.find('Button').prop('disabled')).toBeFalsy();
  });
});
