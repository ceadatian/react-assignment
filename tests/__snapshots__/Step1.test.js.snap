// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step1 component renders correctly 1`] = `ShallowWrapper {}`;
